CSS: Headers and Forums

tiffz opened the pull request:
If it's not too annoying, I will try to do daily CSS PRs for a bit while I try to redo all the site CSS/design some new screens. This change changes some fonts on the site and adjusts forums some. Not 100% sure on the new headers (or the new header font). They are a replacement to the background color headers from before, which I do want to replace. But this could probably use a bit more tweaking still. 

<img width="1271" alt="screenshot 2017-02-14 22 11 25" src="https://cloud.githubusercontent.com/assets/2601991/22962714/93d4d236-f303-11e6-8bb0-279dc139c183.png">


Forums are also responsive now. 

<img width="398" alt="screenshot 2017-02-14 22 17 27" src="https://cloud.githubusercontent.com/assets/2601991/22962724/a56bd7c4-f303-11e6-8675-57b9790ab121.png">

tiffz committed:
Forums main page is responsive.

tiffz committed:
Merge branch 'master' of https://github.com/LikeMyBread/Saylua into css

tiffz committed:
Changed forum/site headers.

NoiSek commented:
Are you looking for daily design critique or just daily code review?

NoiSek commented on saylua/modules/forums/static-source/scss/main.scss:
```diff
@@ -1,17 +1,13 @@
   width: 100%;
-  margin: 0;
-  margin-bottom: 5rem;
+  margin: 1em 0 5em;
```
be explicit with that last one m8y, use all four if you're gonna do the first three.

tiffz replied:
Thanks! I will do this from now on.

NoiSek commented on saylua/modules/forums/static-source/scss/main.scss:
```diff
@@ -25,15 +21,37 @@ ul.forum-main-list {
+      font-weight: bold;
+      letter-spacing: 1px;
+      line-height: 120%;
+      padding: 0.5em .8rem 0.25em;
```
explicitness++

NoiSek commented on saylua/modules/forums/templates/main.html:
```diff
@@ -4,7 +4,19 @@
 {% block body %}
-<h1>Forums</h1>
+<h1>
+  <i class="fa fa-fw fa-comment" aria-hidden="true"></i>
```
Maybe we should give these some alt / title text or something for accessibility.

tiffz replied:
Good call. I wasn't sure if this would be necessary (or desirable) here since the icon is more decorative than functional? Though there are at least a dozen places where I've been lazy about adding title/alt text where it could make sense.

NoiSek commented on saylua/modules/search/templates/results.html:
```diff
@@ -1,7 +1,7 @@
 {% set page_title = "Search Results" %}
 {% block body %}
-<h2>Search Saylua{% if query %}: {{query}}{% endif %}</h2>
+<h1>Search Saylua{% if query %}: {{query}}{% endif %}</h1>
```
`{{ query or "" }}` ?

tiffz replied:
That is a lot cleaner. Thanks!

tiffz replied:
Oh wait, this is not quite as simple since there is the optional ":"

NoiSek replied:
Oh, well there's `{{ ": {}".format(query) if query else "" }}` I suppose. That's not really very far from the original though.

tiffz replied:
I like that though. It's more readable.

NoiSek commented on saylua/static-source/scss/layout/_layout.scss:
```diff
@@ -39,58 +40,81 @@
+    z-index: 50;
     bottom: -1rem;
     left: -1rem;
-    z-index: 50;
```
ahueahuehaehauhee z-index wars

NoiSek commented on saylua/templates/layout.html:
```diff
@@ -40,59 +41,62 @@
+            {% if category == 'error' %}
+              <div class="notification error">Error: {{ message }} <a class="close-button" href="#" title="close">&#x00d7;</a></div>
+            {% else %}
+              <div class="notification">{{ message }} <a class="close-button" href="#" title="close">&#x00d7;</a></div>
```
These two should really just be condensed together into one notification with variable class names. That predates this PR though and is unrelated.

tiffz replied:
Easy to do though, and might as well. Done!

NoiSek reviewed (changes requested):
Gimme dem dere explicit properties if you're gonna use the padding / margin shorthand. (If you're gonna use the margin shorthand and use three properties instead of two, that is)

Didn't comment on every occurrence but there are a few of these.

tiffz committed:
Addressed comments and some random combing.

tiffz commented:
Addressed comments. I tried to grab as many of the unexplicit margin/padding declarations as I could, but I probably missed some. I will add that to my journey. :D 

I was mostly thinking of daily code review so that I don't end up turning the CSS codebase into anything too unmaintainable at once. 

I would be extremely delighted to receive any design critique as well though. :D

tiffz committed:
Simplified forums CSS some.

tiffz committed:
Minor spacing tweaks.

tiffz committed:
Added a random_pet_image() template function and some other stuff.

tiffz commented:
Tweaked the design some more. 

<img width="1279" alt="screenshot 2017-02-15 23 41 29" src="https://cloud.githubusercontent.com/assets/2601991/23011956/98687228-f3d8-11e6-9e77-c1900f6f00fe.png">

tiffz committed:
Made if query simpler.

NoiSek commented on saylua/provisioner.py:
```diff
@@ -6,17 +6,17 @@
     display_name = "admin"
     username = "admin"
-    phash = User.hash_password("password") #Yes, the default password is password
+    phash = User.hash_password("password")  # Yes, the default password is password
```
++

NoiSek commented on saylua/provisioner.py:
```diff
@@ -67,18 +68,16 @@ def setup():
             display_name = soulname(7)
             username = display_name
-            phash = User.hash_password("password") #Yes, the default password is password
+            phash = User.hash_password("password")  # Yes, the default password is password
```
+++

NoiSek commented on saylua/provisioner.py:
```diff
@@ -67,18 +68,16 @@ def setup():
-        category_keys = []
         for category in categories:
-            category_key = BoardCategory(title=category).put().urlsafe() #Keys currently in old form
+            category_key = BoardCategory(title=category).put().urlsafe()  # Keys currently in old form
```
++++

NoiSek reviewed (approved):
a y y
